examples/shor.py

```diff
@@ -83,9 +95,179 @@ def naive_order_finder(x: int, n: int) -> int:
+                else:
+                    wire_symbols.append('a' + str(a))
+                a += 1
+            if qubit in self.exponent_register:
+                wire_symbols.append('e' + str(e))
+                e += 1
+        return cirq.CircuitDiagramInfo(wire_symbols=tuple(wire_symbols))
+
+
+def bitlength(n: int) -> int:
+    """Returns minimum number of bits needed to specify a positive integer."""
+    return int(math.ceil(math.log(n) / math.log(2)))
+
+
+def make_order_finding_circuit(x: int, n: int) -> cirq.Circuit:
+    """Returns quantum circuit which computes the order of x modulo n.
+
+    The circuit uses Quantum Phase Estimation to compute an eigenvalue of
+    the unitary
+
+        U|y⟩ = |y * x mod n⟩      0 <= y < n
+        U|y⟩ = |y⟩                n <= y
+
+    discussed in the header of this file. The circuit uses two registers:
+    the ancilla register which is acted on by U and the exponent register
+    from which an eigenvalue is read out after measurement at the end. The
+    circuit consists of three steps:
+
+    1. Initialization of the ancilla register to |0..01⟩ and the exponent
+       register to a superposition state.
+    2. Multiple controlled-U^2^j operations implemented efficiently using
+       modular exponentiation.
+    3. Inverse Quantum Fourier Transform to kick an eigenvalue to the
+       exponent register.
+
+    Args:
+        x: positive integer whose order modulo n is to be found
+        n: modulus relative to which the order of x is to be found
+
+    Returns:
+        Quantum circuit for finding the order of x modulo n
+    """
+    L = bitlength(n)
+    ancilla_register = cirq.LineQubit.range(L)
+    exponent_register = cirq.LineQubit.range(L, 3 * L + 3)
+    return cirq.Circuit(
+        cirq.X(ancilla_register[L - 1]),
+        cirq.H.on_each(*exponent_register),
+        ModularExp(ancilla_register, exponent_register, x, n),
+        cirq.QFT(*exponent_register, inverse=True),
+        cirq.measure(*exponent_register, key='exponent'),
+    )
+
+
+def read_eigenphase(result: cirq.TrialResult) -> float:
+    """Interprets the output of the order finding circuit.
+
+    Specifically, it returns s/r such that exp(2πis/r) is an eigenvalue
+    of the unitary
+
+        U|y⟩ = |xy mod n⟩  0 <= y < n
+        U|y⟩ = |y⟩         n <= y
+
+    described in the header of this file.
+
+    Args:
+        result: trial result obtained by sampling the output of the
+            circuit built by make_order_finding_circuit
+
+    Returns:
+        s/r where r is the order of x modulo n and s is in [0..r-1].
+    """
+    exponent_as_integer = result.data['exponent'][0]
+    exponent_num_bits = result.measurements['exponent'].shape[1]
+    return float(exponent_as_integer / 2**exponent_num_bits)
+
+
+def quantum_order_finder(x: int, n: int) -> Optional[int]:
+    """Computes smallest positive r such that x^r mod n == 1.
```

Comment:
Use python notation for equations. ** instead of ^

Reply:
Done throughout.